sound/soc/sof/sof-client.c

```diff
@@ -0,0 +1,242 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
+//
+#include <linux/device.h>
+#include "sof-priv.h"
+#include "sof-client.h"
+#include "ops.h"
+
+void *sof_get_client_data(struct device *dev)
+{
+	struct snd_sof_client *client = dev_get_platdata(dev);
+
+	return client->client_data;
+}
+EXPORT_SYMBOL(sof_get_client_data);
+
+const struct sof_dev_desc *sof_get_dev_desc(struct device *dev)
+{
+	struct snd_sof_dev *sdev = dev_get_drvdata(dev->parent);
+	struct snd_sof_pdata *sof_pdata = sdev->pdata;
+
+	return sof_pdata->desc;
+}
+EXPORT_SYMBOL(sof_get_dev_desc);
+
+void sof_client_machine_select(struct device *dev)
+{
+	struct snd_sof_dev *sdev = dev_get_drvdata(dev->parent);
+
+	snd_sof_machine_select(sdev, dev);
+}
+EXPORT_SYMBOL(sof_client_machine_select);
+
+void sof_client_set_mach_params(const struct snd_soc_acpi_mach *mach,
+				struct device *dev)
+{
+	snd_sof_set_mach_params(mach, dev);
+}
+EXPORT_SYMBOL(sof_client_set_mach_params);
+
+struct snd_soc_dai_driver *sof_client_get_dai_drv(struct device *dev)
+{
+	struct snd_sof_dev *sdev = dev_get_drvdata(dev->parent);
+
+	return sof_ops(sdev)->drv;
+}
+EXPORT_SYMBOL(sof_client_get_dai_drv);
```

Comment:
Some of these smaller ones could be static inline.

Reply:
sure, will change them

Reply:
@lgirdwood I think I may have spoken too soon here. I actually did consider making them static inline but I ended up not doing it because I wanted to restrict the use of the core ops/definitions ie I want to restrict including the core headers, namely, ops.h and sof-priv.h only to sof-client.c and not the client code. Is there a more elagant way to do this?

Reply:
if you want them private thye can still be static inline in ops-priv.h or a similar name.